### PR TITLE
Fancy up release index.html

### PR DESCRIPTION
**Pull request by hpinkos**
This is the first thing users see when they download the Cesium release zip and navigate to `localhost:8080`.  This page needed a bit of an overhaul to give a better first impression.

More feedback as to what we should/should not include on this page is definitely welcome!

Here are the changes I made:
- Added the 'Next Dimension Mapping' tagline under the logo
- Updated styling and broke the page into more well defined sections
- Added section about Cesium ion, encouraging users to sign up for access to imagery and terrain
- Added links to cesium.com and the blog in the 'External links' section

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @hpinkos!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/release-index-html/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by hpinkos** [comment]
Do you think we need links for how to run the unit tests in 6 different ways?  Do you think anyone actually does this?  I was debating just putting one link to run all unit tests under the 'Local links' section.

**Comment by pjcozzi** [comment]
> Do you think we need links for how to run the unit tests in 6 different ways? Do you think anyone actually uses these?

For the dev version, yes, we need them.  For release, no.  Good call.

> I was debating just putting one link to run all unit tests under the 'Local links' section.

OK

**Comment by pjcozzi** [comment]
Please bump to me for a quick final look after @mramato reviews.

**Comment by hpinkos** [comment]
@OmarShehata the save/load thing sounds like it might be a good candidate for a Sandcastle example if we don't have one already.  I agree that's a question that comes up pretty often.

File: index.release.html [hunk 1]
```diff
@@ -1,116 +1,140 @@
...
+            </tr>
+            </tbody>
+        </table>
+    </div>
+    <div class="section">
+        <h2>External links</h2>
+        <table>
+            <tbody>
+            <tr>
+                <td><a href="//cesium.com">Cesium</a></td>
```
Comment (emackey) [inline index.release.html]: I think all the leading `//` should change to `https://` at this point.

Comment (hpinkos) [reply index.release.html]: 👍 cool, I wasn't quite sure about that

Comment (emackey) [reply index.release.html]: At some point many years ago, there was an effort to support users on both `http` and `https`, and preserve whichever protocol they're using.  But these days, best practice is to always upgrade to `https` anytime the server offers it.

Comment (emackey) [reply index.release.html]: (which is not the case for webglreport, sadly, it's still http)

**Pull request merged**